src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs

```diff
@@ -126,9 +126,12 @@ private unsafe void FlushInternal(bool endOfRequest, ArraySegment<byte> data = n
             var flags = ComputeLeftToWrite(data.Count, endOfRequest);
             if (endOfRequest && _leftToWrite > 0)
             {
+                if (!RequestContext.DisconnectToken.IsCancellationRequested)
+                {
+                    // This is logged rather than thrown because it is too late for an exception to be visible in user code.
+                    LogHelper.LogError(Logger, "ResponseStream::Dispose", "Fewer bytes were written than were specified in the Content-Length.");
```

Comment:
@Kahbazi can you write a unit test to verify this behavior? Or is there a race and it makes it hard to write a test.

Reply:
I plan to try some manual verification before merging this. I don't know if DisconnectToken.IsCancellationRequested is consistent enough in these scenarios for this to work as expected.